src/Cone3d.elm

```diff
@@ -0,0 +1,391 @@
+        , length = Quantity.abs (arguments.tip |> Quantity.minus arguments.base)
+        }
+
+
+{-| Construct a cone given the center point on the base, axial direction
+towards the tip, radius and length. Negative values for radius or length
+will be treated as positive (the absolute values will be used).
+-}
+startingAt :
+    Point3d units coordinates
+    -> Direction3d coordinates
+    -> { radius : Quantity Float units, length : Quantity Float units }
+    -> Cone3d units coordinates
+startingAt givenBasePoint givenDirection arguments =
+    Types.Cone3d
+        { axis = Axis3d.through givenBasePoint givenDirection
+        , radius = Quantity.abs arguments.radius
+        , length = Quantity.abs arguments.length
+        }
+
+
+{-| Attempt to construct a cone from the given center point on the base,
+tip point and radius. If the base and tip points are coincident (the same point),
+returns `Nothing`.
+-}
+from : Point3d units coordinates -> Point3d units coordinates -> Quantity Float units -> Maybe (Cone3d units coordinates)
+from givenBasePoint givenTipPoint givenRadius =
+    case Direction3d.from givenBasePoint givenTipPoint of
+        Just computedDirection ->
+            Just <|
+                Types.Cone3d
+                    { axis = Axis3d.through givenBasePoint computedDirection
+                    , radius = Quantity.abs givenRadius
+                    , length = Point3d.distanceFrom givenBasePoint givenTipPoint
+                    }
+
+        Nothing ->
+            Nothing
+
+
+{-| Get the central axis of a cone. The origin point of the axis will be at
+the center point of the base, and the direction of the axis will be from the
+cone's base point towards its tip.
+-}
+axis : Cone3d units coordinates -> Axis3d units coordinates
+axis (Types.Cone3d cone) =
+    cone.axis
+
+
+{-| Get the axial direction of a cone.
+-}
+axialDirection : Cone3d units coordinates -> Direction3d coordinates
+axialDirection cone =
+    Axis3d.direction (axis cone)
+
+
+{-| Get the base radius of a cone.
+-}
+radius : Cone3d units coordinates -> Quantity Float units
+radius (Types.Cone3d cone) =
+    cone.radius
+
+
+{-| Get the base diameter of a cone (twice its radius).
+-}
+diameter : Cone3d units coordinates -> Quantity Float units
+diameter cone =
+    Quantity.twice (radius cone)
+
+
+{-| Get the overall length of a cone.
+-}
+length : Cone3d units coordinates -> Quantity Float units
+length (Types.Cone3d cone) =
+    cone.length
+
+
+{-| Get the base point of a cone. This is the center point of the circle
+that forms the base of the cone.
+-}
+basePoint : Cone3d units coordinates -> Point3d units coordinates
+basePoint cone =
+    Point3d.along (axis cone) (Quantity.multiplyBy -0.5 (length cone))
+
+
+{-| Get the tip point of a cone.
+-}
+tipPoint : Cone3d units coordinates -> Point3d units coordinates
+tipPoint cone =
+    Point3d.along (axis cone) (Quantity.multiplyBy 0.5 (length cone))
```

Comment:
Similarly, I think this should now be `Point3d.along (axis cone) (length cone)`

Reply:
Thanks, fixed!